test/extended/builds/pipeline.go

```diff
@@ -115,10 +150,10 @@ var _ = g.Describe("[Feature:Builds][Slow] openshift pipeline build", func() {
 			if jenkinsTemplatePath == jenkinsPersistentTemplatePath {
 				jenkinsTemplateName = "jenkins-persistent"
 
-				err = exutil.AddNamespaceLabelToPersistentVolumeClaimsInTemplate(oc, jenkinsTemplateName)
+				nfspod, pvs, err = exutil.SetupK8SNFSServerAndVolume(oc, 3)
 				o.Expect(err).NotTo(o.HaveOccurred())
 
-				_, err := exutil.SetupNFSBackedPersistentVolumes(oc, "2Gi", 3)
+				err = exutil.AddNamespaceLabelToPersistentVolume(oc, pvs)
```

Comment:
this appears to be the only test calling this, why aren't the others?  (or why is this one?)

Reply:
There are differences in how the jenkins template and the mongo/postgres/mysql templates declare PVCs that had bearing here, at least in my local testing.

The jenkins template explicitly defines a PVC.  Although the jenkins template does not specify this, when the PVCs were actually created, they had a selector where a label with "namespace" as the key was required.  Hence, I had to add the label to the NFS PVs in order for the claim to be bound to one of the PVs.

The other templates do not define a PVC explicitly.  Instead, the define things like a `volumeClaimTemplates`.  Those resulted in PVCs without this specific selector, and the PVs were matched without any need to update them.

Now, haven't obtained a lean image eco run yet, though they failed before the ext tests actually ran.

Curious to see if the PVC to PV matching works out like it did for me locally.  Also, by locally, I mean running the ext tests where the ext tests provision the cluster.  I moved to that model when I saw the NFS image's mounts were not properly established without may latest update to that custom image we have.

Reply:
Those `exutil.AddNamespaceLabelToPersistentVolumeClaimsInTemplate` calls were failing of late, hence the change of gears you see here.

But yes, as per the first actual test run in the PR we got, sorting out these details across the different templates/db types needs some fine tuning, as per my earlier note I suspected they would.

I'm still inclined to update the PVC after its creation vs. trying to patch the various templates, given the variable nature I saw with `exutil.AddNamespaceLabelToPersistentVolumeClaimsInTemplate`, but of course that can change based on what I see as I work through them.

Will add some comments to clarify direction, etc. on my next commit.

Reply:
i think patching the PVC after creating it is a mistake.  Once you've created the PVC, it's going to get bound to a PV immediately.  I'm not sure what effect changing it after will have, but i suspect it will be too late to accomplish what you want.

Reply:
that is assuming PVs are already present when the PVC is created 

in any event, sorting through these machinations is the top of my to-do list, as part of sorting through the results of a combined run

Reply:
OK I just ran mongog/postgres/jenkins together in 1 single run as was able to bind pvcs without patching templates nor labelling PVs, and I saw no additional PVs getting created now by the extended test framework.

So I'm commenting out the labelling in pipeline.go, at least for now.

In case there is some difference between what occurs when launching the extended tests locally vs. in our ci jobs around PVs, I've added a series of PV/PVC dumps before and after setup in each test, to expose that if it is the case, where perhaps that would explain the hang on bind in say the postgres run after the mongdob run completed.

Will be pushing these changes and starting another run momentarily.